File: public/cses/sorting-and-searching/ferris-wheel/approach/ferris-wheel.py
# setup-start
import os
import sys
script_directory = os.path.dirname(os.path.realpath(__file__))
parent_directory = os.path.dirname(script_directory)
input_path = os.path.join(parent_directory, 'test_cases/1/input.txt')
output_path = os.path.join(parent_directory, 'test_cases/1/output.txt')
if os.path.exists(input_path):
    sys.stdin = open(input_path, 'r')
    try:
        sys.stdout = open(output_path, 'w')
    except Exception as e:
        pass
# setup-end


def main():
    n, x = map(int, input().split())
    p = list(map(int, input().split()))

    p.sort()
    l, r = 0, n - 1
    ans = 0

    while l <= r:
        if p[l] + p[r] <= x:
            ans += 1
            l += 1
            r -= 1
        else:
            r -= 1
            ans += 1

    print(ans)


main()

File: public/cses/sorting-and-searching/ferris-wheel/approach/ferris-wheel.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // setup
    // setup-start
#ifndef ONLINE_JUDGE
    freopen("../test_cases/1/input.txt", "r", stdin);
    freopen("../test_cases/1/output.txt", "w", stdout);
#endif
    // setup-end

    int n, x;
    cin >> n >> x;
    vector<int> p(n);
    for (int i = 0; i < n; i++) {
        cin >> p[i];
    }
    sort(p.begin(), p.end());
    int l = 0;
    int r = n - 1;
    int ans = 0;
    while (l <= r) {
        if (p[l] + p[r] <= x) {
            ans += 1;
            l += 1;
            r -= 1;
        } else {
            r -= 1;
            ans += 1;
        }
    }
    cout << ans << endl;
}